app/src/main/java/org/metabrainz/android/presentation/features/userprofile/ProfileActivity.kt

```diff
@@ -0,0 +1,104 @@
+package org.metabrainz.android.presentation.features.userprofile
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import com.google.android.material.tabs.TabLayoutMediator
+import org.metabrainz.android.R
+import org.metabrainz.android.databinding.ActivityProfileBinding
+import org.metabrainz.android.presentation.features.dashboard.DashboardActivity
+import org.metabrainz.android.presentation.features.login.LoginSharedPreferences
+import org.metabrainz.android.presentation.features.settings.SettingsActivity
+
+class ProfileActivity : AppCompatActivity() {
+    lateinit var binding: ActivityProfileBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding= ActivityProfileBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        binding.profileTopAppBar.setContent {
+            TopAppBar()
+        }
+
+        val pagerAdapter= ProfileViewPagerAdapter(this)
+        binding.profilePager.adapter = pagerAdapter
+        TabLayoutMediator(binding.profileTabLayout, binding.profilePager){tab,position->
+            run{
+                when(position){
+                    0->tab.text = "Profile"
+                    1->tab.text = "Subscriptions"
+                    2->tab.text = "Subscribers"
+                    3->tab.text = "Collections"
+                    4->tab.text = "Tags"
+                    5->tab.text = "Ratings"
+                }
+            }
+
+        }.attach()
+    }
+
+    private fun logoutUser() {
+        LoginSharedPreferences.logoutUser()
```

Comment:
Can we have an alert shown first that this would logout the user?

Reply:
Alert Dialog Box will be popped everytime we try to log out